hail/python/hailtop/batch_client/aioclient.py

```diff
@@ -535,7 +535,7 @@ def _create_job(self,
                     in_update_parent_ids.append(job._job_id)
             else:
                 assert isinstance(job, SubmittedJob)
-                if self._batch is None or job._batch != self._batch:
+                if self._batch is None or job._batch.id != self._batch.id:
```

Comment:
I'm concerned that multiple `Batch` objects that represent the same batch floating around in memory could cause difficult to diagnose bugs surrounding equality. Is the reason for this change because the `_run` function accepts a `Batch` and then returns a different `Batch` to the user? Could we without much additional effort ensure that only one `Batch` object exists after resubmission? Having the number of batch objects be `O(updates)` feels like a memory leak (though not a big one).

Reply:
Yes that was the reason. Hopefully, I fixed it in a reasonable way. I also added a static method to create a Batch from a batch id that I used for Siwei's use case.

Reply:
This should be reverted such that the assertion is `job._batch != self._batch`